packages/assets-controllers/src/TokenListController.ts

```diff
@@ -156,75 +144,26 @@ export class TokenListController extends StaticIntervalPollingController<TokenLi
     });
     this.intervalDelay = interval;
     this.cacheRefreshThreshold = cacheRefreshThreshold;
-    this.chainId = chainId;
     this.updatePreventPollingOnNetworkRestart(preventPollingOnNetworkRestart);
     this.abortController = new AbortController();
-    if (onNetworkStateChange) {
-      // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-      // eslint-disable-next-line @typescript-eslint/no-misused-promises
-      onNetworkStateChange(async (networkControllerState) => {
-        await this.#onNetworkControllerStateChange(networkControllerState);
-      });
-    } else {
-      this.messagingSystem.subscribe(
-        'NetworkController:stateChange',
-        // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-        // eslint-disable-next-line @typescript-eslint/no-misused-promises
-        async (networkControllerState) => {
-          await this.#onNetworkControllerStateChange(networkControllerState);
-        },
-      );
-    }
-  }
-
-  /**
-   * Updates state and restarts polling on changes to the network controller
-   * state.
-   *
-   * @param networkControllerState - The updated network controller state.
-   */
-  async #onNetworkControllerStateChange(networkControllerState: NetworkState) {
-    const selectedNetworkClient = this.messagingSystem.call(
-      'NetworkController:getNetworkClientById',
-      networkControllerState.selectedNetworkClientId,
-    );
-    const { chainId } = selectedNetworkClient.configuration;
-
-    if (this.chainId !== chainId) {
-      this.abortController.abort();
-      this.abortController = new AbortController();
-      this.chainId = chainId;
-      if (this.state.preventPollingOnNetworkRestart) {
-        this.clearingTokenListData();
-      } else {
-        // Ensure tokenList is referencing data from correct network
-        this.update(() => {
-          return {
-            ...this.state,
-            tokenList: this.state.tokensChainsCache[this.chainId]?.data || {},
-          };
-        });
-        await this.restart();
-      }
-    }
```

Comment:
Let's revive this. We still want to update `tokenList` whenever the current network gets changed by the user (we don't want to have to wait for the polling loop to do this).

This would update it immediately